pkg/extensions/cluster_test.go

```diff
@@ -62,6 +64,346 @@ var _ = Describe("Cluster", func() {
+					Name: clusterName,
+				},
+				Spec: extensionsv1alpha1.ClusterSpec{
+					CloudProfile: runtime.RawExtension{
+						Raw: encode(expectedCloudProfile),
+					},
+				},
+			}
+		})
+
+		It("should retrieve cloudprofile from cluster", func() {
+			cloudProfile, err := CloudProfileFromCluster(cluster)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(cloudProfile).To(Equal(expectedCloudProfile))
+		})
+
+		It("should return an error because the cloudprofile cannot be decoded the cluster", func() {
+			cluster.Spec.CloudProfile.Raw = []byte(`{`)
+			Expect(fakeSeedClient.Create(ctx, cluster)).To(Succeed())
+
+			cloudProfile, err := CloudProfileFromCluster(cluster)
+			Expect(err).To(MatchError(ContainSubstring("unexpected end of JSON input")))
+			Expect(cloudProfile).To(BeNil())
+		})
+
+		It("should return an nil because the cloudprofile is not in raw format the cluster", func() {
+			cluster.Spec.CloudProfile.Raw = nil
+
+			cloudProfile, err := CloudProfileFromCluster(cluster)
+			Expect(err).To(BeNil())
+			Expect(cloudProfile).To(BeNil())
+		})
+	})
+
+	Describe("#SeedFromCluster", func() {
+		var (
+			expectedSeed *gardencorev1beta1.Seed
+
+			clusterName string
+			cluster     *extensionsv1alpha1.Cluster
+		)
+
+		BeforeEach(func() {
+			expectedSeed = &gardencorev1beta1.Seed{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "core.gardener.cloud/v1beta1",
+					Kind:       "Seed",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+			}
+
+			clusterName = "foo"
+			cluster = &extensionsv1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: clusterName,
+				},
+				Spec: extensionsv1alpha1.ClusterSpec{
+					Seed: runtime.RawExtension{
+						Raw: encode(expectedSeed),
+					},
+				},
+			}
+		})
+
+		It("should retrieve seed from cluster", func() {
+			seed, err := SeedFromCluster(cluster)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(seed).To(Equal(expectedSeed))
+		})
+
+		It("should return an error because the seed cannot be decoded the cluster", func() {
+			cluster.Spec.Seed.Raw = []byte(`{`)
+			Expect(fakeSeedClient.Create(ctx, cluster)).To(Succeed())
+
+			seed, err := SeedFromCluster(cluster)
+			Expect(err).To(MatchError(ContainSubstring("unexpected end of JSON input")))
+			Expect(seed).To(BeNil())
+		})
+
+		It("should return an nil because the seed is not in raw format the cluster", func() {
+			cluster.Spec.Seed.Raw = nil
+
+			seed, err := SeedFromCluster(cluster)
+			Expect(err).To(BeNil())
+			Expect(seed).To(BeNil())
+		})
+	})
+
+	Describe("#ShootFromCluster", func() {
+		var (
+			expectedShoot *gardencorev1beta1.Shoot
+
+			clusterName string
+			cluster     *extensionsv1alpha1.Cluster
+		)
+
+		BeforeEach(func() {
+			expectedShoot = &gardencorev1beta1.Shoot{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "core.gardener.cloud/v1beta1",
+					Kind:       "Shoot",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "foo",
+				},
+			}
+
+			clusterName = "foo"
```

Comment:
No need to assign the variable, instead use it directly. As this is not used anywhere else.